source/translated_table_model.ts

```diff
@@ -168,11 +168,13 @@ export class TranslatedTableModel extends TableModel {
     this.beginTransaction();
     const operationIndex = this.reverseTranslation[operation.index];
     const end = this.translation.length - 1;
```

Comment:
Redo this method. Look to consolidate as many loops as possible into a single loop.